File: vivy_encrytpor/src/main/java/com/vivy/asymmetric/AsymmetricEncryption.java
package com.vivy.asymmetric;

import java.security.PrivateKey;
import java.security.PublicKey;

public interface AsymmetricEncryption {

  String encryptText(PublicKey publicKey, String decryptedText);

  String decryptText(PrivateKey privateKey, String base64AndEncryptedContent);
}

File: vivy_encryptor/src/main/java/com/vivy/asymmetric/AsymmetricEncryption.kt
package com.vivy.asymmetric

import java.security.PrivateKey
import java.security.PublicKey

interface AsymmetricEncryption {

    fun encryptText(
        publicKey: PublicKey,
        decryptedText: String
    ): String

    fun decryptText(
        privateKey: PrivateKey,
        base64AndEncryptedContent: String
    ): String
}
